Makefile

```diff
@@ -3,41 +3,39 @@ DIST := skygear-server
 VERSION := $(shell git describe --always)
 GIT_SHA := $(shell git rev-parse HEAD)
 BUILD_DATE := $(shell date -u +"%Y-%m-%dT%H:%M:%SZ")
+TARGETS := gateway auth migrate
 GO_BUILD_LDFLAGS := -ldflags "-X github.com/skygeario/skygear-server/pkg/server/skyversion.version=$(VERSION)"
 GO_TEST_TIMEOUT := 1m30s
 OSARCHS := linux/amd64 linux/386 linux/arm windows/amd64 windows/386 darwin/amd64
 GO_TEST_CPU := 1,4
-GO_TEST_PACKAGE := ./pkg/core/... ./pkg/auth/... ./pkg/gateway/... ./pkg/record/...
+GO_TEST_PACKAGE := ./pkg/core/... ./pkg/auth/... ./pkg/gateway/...
 SHELL := /bin/bash
 
-ifeq (1,${WITH_ZMQ})
-GO_BUILD_TAGS := --tags zmq
-endif
-
 ifeq (1,${GO_TEST_VERBOSE})
 GO_TEST_ARGS_VERBOSE := -v
 endif
 
-DOCKER_COMPOSE_CMD := docker-compose \
-	-f docker-compose.make.yml \
+DOCKER_COMPOSE_CMD := docker-compose -f docker-compose.make.yml
 
-DOCKER_COMPOSE_CMD_TEST := docker-compose \
-	-f docker-compose.test.yml \
-	-p skygear-server-test
-
-ifeq (1,${WITH_DOCKER})
+ifeq (1,${WITH_DOCKER})	
 DOCKER_RUN := ${DOCKER_COMPOSE_CMD} run --rm app
-DOCKER_RUN_DB := ${DOCKER_COMPOSE_CMD_TEST} run --rm db_cmd
-DOCKER_RUN_TEST := ${DOCKER_COMPOSE_CMD_TEST} run --rm app
-GO_TEST_TIMEOUT := 5m
 endif
 
 DOCKER_REGISTRY :=
 DOCKER_ORG_NAME := skygeario
-DOCKER_IMAGE := skygear-server
+DOCKER_IMAGE_AUTH := skygear-auth
+DOCKER_IMAGE_GATEWAY := skygear-gateway
+DOCKER_IMAGE_MIGRATE := skygear-migrate
 DOCKER_TAG := git-$(shell git rev-parse --short HEAD)
 PUSH_DOCKER_TAG := $(VERSION)
-IMAGE_NAME := $(DOCKER_REGISTRY)$(DOCKER_ORG_NAME)/$(DOCKER_IMAGE):$(DOCKER_TAG)
+IMAGE_NAME = $(DOCKER_REGISTRY)$(DOCKER_ORG_NAME)/$(1):$(DOCKER_TAG)
+VERSIONED_IMAGE_NAME = $($(DOCKER_REGISTRY)$(DOCKER_ORG_NAME)/$(1):$(PUSH_DOCKER_TAG))
```

Comment:
should be `$(DOCKER_REGISTRY)$(DOCKER_ORG_NAME)/$(1):$(PUSH_DOCKER_TAG)`? Now `VERSIONED_IMAGE_NAME` is empty

Reply:
@carmenlau Thanks. Updated.